letter-sender/src/main/java/com/christmas/letter/sender/exceptions/LetterValidator.java

```diff
@@ -0,0 +1,25 @@
+package com.christmas.letter.sender.exceptions;
+
+import com.christmas.letter.sender.model.Letter;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LetterValidator {
+
+  public void validate(Letter letter){
```

Comment:
This isn't used, however, I suggest 2 possible approaches here - reference for the future, no need to waste time on it unless you want to. 
1. I would create an annotation, for example @ValidCoordinate or something similar, and I would move this code in a validator linked to that annotation. If you want to go this path and can't find an example, let me know.
2. I would simplify how we store the coordinates - make it a string and store the address instead, or still make it a string and use the above recommendation.  

And if we go this path, should we allow negative values for example?

Reply:
I called this validator method in service method.
on mape are some locations where the coordinates are negative, example: 21.3069°, -157.8583°

Reply:
This would be a cleaner way to do the validation: https://www.baeldung.com/spring-mvc-custom-validator